### PR TITLE
Creates another script to back populate the user_proficiencies table.

### PR DESCRIPTION
**Pull request by ashercodeorg**

### CODE REVIEW

**Comment by ashercodeorg** [comment]
Ping. Looking to run this soon, as the data has been generated.

File: bin/oneoff/back_populate_user_proficiencies_again [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+
+  (1..50).each do |index|
+    sum[index] += incremental_line[index]
+  end
+
+  unless incremental_line[51].nil?
+    sum[51] = incremental_line[51]
+  end
+
+  if sum[52] && sum.proficient?
```
Comment (Hamms) [inline bin/oneoff/back_populate_user_proficiencies_again]: shouldn't this be `proficient?(sum)`?

File: bin/oneoff/back_populate_user_proficiencies_again [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+    user_proficiency_hash[user_id] = increment_user_proficiency(historical, line)
+  end
+
+  puts "PROCESSED #{filename}..."
+end
+
+hash_count = 0
+puts "UPDATING DB..."
+# Update the DB with the data in user_proficiency_hash.
+user_proficiency_hash.each do |user_id, proficiency|
```
Comment (Hamms) [inline bin/oneoff/back_populate_user_proficiencies_again]: are we going to have enough of these to warrant batching?

File: bin/oneoff/back_populate_user_proficiencies_again (hunk 1 above)
Comment (ashercodeorg) [reply bin/oneoff/back_populate_user_proficiencies_again]: Oops. Nice catch. Actually, it should be `if historical_line[52].nil? && proficient?(sum)`.

**Comment by Hamms** [comment]
LGTM!

File: bin/oneoff/back_populate_user_proficiencies_again (hunk 2 above)
Comment (ashercodeorg) [reply bin/oneoff/back_populate_user_proficiencies_again]: It certainly can't hurt. Let me wrap this in a transaction that processes 1K records at a time.

File: bin/oneoff/back_populate_user_proficiencies_again [hunk 3]
```diff
@@ -0,0 +1,357 @@
...
+  CSV.foreach(filename).with_index do |line, line_num|
+    if line_num % 100_000 == 0
+      puts "  LINE #{line_num}..."
+    end
+
+    # Convert the user_id and level counts from strings to integers.
+    (0..50).each do |index|
+      line[index] = line[index].to_i
+    end
+    # Add a nil value for basic_proficiency_at.
```
Comment (ashercodeorg) [inline bin/oneoff/back_populate_user_proficiencies_again:261]: Index 0: user_id
Index 1-50: level_counts
Index 51: last_progress_at (for the month)
Index 52: basic_proficiency_at

**Comment by ashercodeorg** [comment]
Version merged reflects the version run in production. It took a few minutes to process the data and (extrapolating) a bunch of hours (eight?) to do the DB writes.

**Pull request merged**